### PR TITLE
Marks

### PR DESCRIPTION
**Pull request by ghankerson**

### CODE REVIEW

File: src/lib/components/Body/test/MultipleMarks.test.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import React from "react";
```
Comment (etherbob) [inline src/lib/components/Body/test/MultipleMarks.test.js]: Is this file still needed?

Comment (ghankerson) [reply src/lib/components/Body/test/MultipleMarks.test.js]: The first test is a real bug that we patched in this commit  so it should stay I think.   The convertMarks stuff can go

Comment (ghankerson) [reply src/lib/components/Body/test/MultipleMarks.test.js]: Removed convertMarks and related unused tests

File: src/lib/utils/Mark.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import Dispatch from "./Dispatch";
+import uuid from "uuid";
+
+const Mark = (mark, inner, props) => {
+  const Dispatcher = Dispatch(mark.type);
+  return (
+    <Dispatcher
+      key={uuid()}
+      nodeData={mark}
```
Comment (etherbob) [inline src/lib/utils/Mark.js]: Would markData be clearer? Or, alternatively I kind of liked how you passed in the attrs themselves as a prop before. I don't think the mark handlers need any other mark info beside attrs.

File: src/lib/components/Body/test/MultipleMarks.test.js (hunk 1 above)
Comment (etherbob) [inline src/lib/components/Body/test/MultipleMarks.test.js]: Is this still needed?

File: src/lib/utils/Traverse.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      <ApmAttachment
+        nodeData={props.nodeData}
+        embedded={props.embedded}
+        key={uuid()}
+      />
+    );
+  }
+
+  if (props.nodeData.content) {
+    for (let i = 0; i < props.nodeData.content.length; ++i) {
```
Comment (etherbob) [inline src/lib/utils/Traverse.js]: could this be a forEach() or map()?

File: src/lib/components/Body/test/MultipleMarks.test.js (hunk 1 above)
Comment (ghankerson) [reply src/lib/components/Body/test/MultipleMarks.test.js]: Yes removed  most of it in this commit [fded561d0033d10bcc95dcb0f419bf741995d0ee](https://github.com/APMG/amat-react/pull/7/commits/fded561d0033d10bcc95dcb0f419bf741995d0ee)

File: src/lib/utils/Traverse.js (hunk 3 above)
Comment (ghankerson) [reply src/lib/utils/Traverse.js]: No not as is I need that break statement and there is not one for forEach or Map. I'll see if I can clean up somehow

File: src/lib/utils/Mark.js (hunk 2 above)
Comment (ghankerson) [reply src/lib/utils/Mark.js]: Addressed in e68239b62b83bd87

File: src/lib/utils/Traverse.js (hunk 3 above)
Comment (ghankerson) [reply src/lib/utils/Traverse.js]: Don't really have a solution for this right now  -  do you mind if I ticket it  with  other refactoring for Traverse?

File: src/lib/utils/Mark.js (hunk 2 above)
Comment (etherbob) [reply src/lib/utils/Mark.js]: Ooh, the only caution I have there is what happens if a mark gets an attr of key or inner?

File: src/lib/utils/Traverse.js (hunk 3 above)
Comment (etherbob) [reply src/lib/utils/Traverse.js]: Sounds good. Is this mostly to cover the ApmTableOfContents use case?

File: src/lib/utils/Mark.js (hunk 2 above)
Comment (ghankerson) [reply src/lib/utils/Mark.js]: key is kind of informally namespace for React   should I throw an Error "Illegal attribute" 

don't really have another idea ...

File: src/lib/utils/Traverse.js (hunk 3 above)
Comment (ghankerson) [reply src/lib/utils/Traverse.js]: Yes

File: src/lib/utils/Mark.js (hunk 2 above)
Comment (ghankerson) [reply src/lib/utils/Mark.js]: Addressed in ae6bf92

**Comment by etherbob** [comment]
Looks great.
